file_browser/src/components/ActionBar.tsx

```diff
@@ -0,0 +1,73 @@
+import { useState } from "react";
+import { observer } from "mobx-react-lite";
+import fileSystemStore from "../stores/FileSystemStore";
+import { FileClass } from "../classes/File";
+import AddDialog from "./AddDIalog";
+import RenameDialog from "./RenameDialog";
+import EditDialog from "./EditDialog";
+import s from "./ActionBar.module.scss";
+
+// icons
+import deleteIcon from "../assets/delete.png";
+import plus from "../assets/plus.png";
+import edit from "../assets/edit.png";
+import renameIcon from "../assets/rename.png";
+
+const ActionBar = observer(() => {
+  const [isRenameDialogOpen, setIsRenameDialogOpen] = useState(false);
+  const [isEditDialogOpen, setIsEditDialogOpen] = useState(false);
+  const [isAddDialogOpen, setIsAddDialogOpen] = useState(false);
```

Comment:
You always set those 3 states together, and there's always only one being true. I recommend using one with an enum that has `none, rename, edit, add` states